api/clients/disperser_client.go

```diff
@@ -37,47 +38,73 @@ func NewConfig(hostname, port string, timeout time.Duration, useSecureGrpcFlag b
 }
 
 type DisperserClient interface {
+	Close() error
 	DisperseBlob(ctx context.Context, data []byte, customQuorums []uint8) (*disperser.BlobStatus, []byte, error)
 	DisperseBlobAuthenticated(ctx context.Context, data []byte, customQuorums []uint8) (*disperser.BlobStatus, []byte, error)
 	GetBlobStatus(ctx context.Context, key []byte) (*disperser_rpc.BlobStatusReply, error)
 	RetrieveBlob(ctx context.Context, batchHeaderHash []byte, blobIndex uint32) ([]byte, error)
 }
 
+// See the NewDisperserClient constructor's documentation for details and usage examples.
 type disperserClient struct {
 	config *Config
 	signer core.BlobRequestSigner
+	// conn and client are not initialized in the constructor, but are initialized lazily
+	// whenever a method is called, using initOnce to make sure initialization happens only once
+	// and is thread-safe
+	initOnce sync.Once
+	conn     *grpc.ClientConn
+	client   disperser_rpc.DisperserClient
 }
 
 var _ DisperserClient = &disperserClient{}
 
-func NewDisperserClient(config *Config, signer core.BlobRequestSigner) DisperserClient {
+// DisperserClient maintains a single underlying grpc connection to the disperser server,
+// through which it sends requests to disperse blobs, get blob status, and retrieve blobs.
+// The connection is established lazily on the first method call. Don't forget to call Close(),
+// which is safe to call even if the connection was never established.
+//
+// DisperserClient is safe to be used concurrently by multiple goroutines.
+//
+// Example usage:
+//
+//	client := NewDisperserClient(config, signer)
+//	defer client.Close()
+//
+//	// The connection will be established on the first call
+//	status, requestId, err := client.DisperseBlob(ctx, someData, someQuorums)
+//	if err != nil {
+//	    // Handle error
+//	}
+//
+//	// Subsequent calls will use the existing connection
+//	status2, requestId2, err := client.DisperseBlob(ctx, otherData, otherQuorums)
+func NewDisperserClient(config *Config, signer core.BlobRequestSigner) *disperserClient {
```

Comment:
Why did we switch the return type to struct pointer vs. interface?

Reply:
https://en.m.wikipedia.org/wiki/Robustness_principle instantiated in golang is "take interfaces as input, and return structs" (see https://bryanftan.medium.com/accept-interfaces-return-structs-in-go-d4cab29a301b for eg).

This is not a breaking change because the pointer implements the interface.

Reply:
This is a good read. Thanks!